javalib/src/main/scala/java/math/MathContext.scala

```diff
@@ -0,0 +1,206 @@
+
+    (precision, roundingMode)
+  }
+
+  private def invalidMathContext(reason: String, s: String): IllegalArgumentException = {
+    throw new IllegalArgumentException(reason + ": " + s)
+  }
+
+  private def checkNotNull(reference: AnyRef, errorMessage: AnyRef): Unit =
+    if (reference == null)
+      throw new NullPointerException(String.valueOf(errorMessage))
+
+}
+
+/**
+ *  Immutable objects describing settings such as rounding mode and digit
+ *  precision for the numerical operations provided by class {@link BigDecimal}.
+ *
+ *  Constructs a new {@code MathContext} with the specified precision and
+ *  with the specified rounding mode. If the precision passed is zero, then
+ *  this implies that the computations have to be performed exact, the
+ *  rounding mode in this case is irrelevant.
+ *
+ *  @param precision
+ *            the precision for the new {@code MathContext}.
+ *  @param roundingMode
+ *            the rounding mode for the new {@code MathContext}.
+ */
+case class MathContext(precision: Int, roundingMode: RoundingMode) extends Serializable {
+
+  checkValid()
+
+  def getPrecision: Int = precision
+
+  def getRoundingMode: RoundingMode = roundingMode
+
+  /**
+   * Constructs a new {@code MathContext} with the specified precision and
+   * with the rounding mode {@link RoundingMode#HALF_UP HALF_UP}. If the
+   * precision passed is zero, then this implies that the computations have to
+   * be performed exact, the rounding mode in this case is irrelevant.
+   *
+   * @param precision
+   *            the precision for the new {@code MathContext}.
+   */
+  def this(precision: Int) {
+    this(precision, RoundingMode.HALF_UP)
+  }
+
+  private def this(args:(Int, RoundingMode)) {
+    this(args._1, args._2)
+  }
+
+  /**
+   * Constructs a new {@code MathContext} from a string. The string has to
+   * specify the precision and the rounding mode to be used and has to follow
+   * the following syntax: "precision=&lt;precision&gt; roundingMode=&lt;roundingMode&gt;"
+   * This is the same form as the one returned by the {@link #toString}
+   * method.
+   *
+   * @throws IllegalArgumentException
+     *             if the string is not in the correct format or if the
+   *             precision specified is < 0.
+   */
+  def this(s: String) {
+    this(MathContext.getArgs(s))
+    checkValid()
+  }
+
+  /**
+   * Returns true if x is a {@code MathContext} with the same precision
+   * setting and the same rounding mode as this {@code MathContext} instance.
+   *
+   * @param x
+     *            object to be compared.
+   * @return {@code true} if this {@code MathContext} instance is equal to the
+   *         {@code x} argument; {@code false} otherwise.
+   */
+  override def equals(x: Any): Boolean =
+    x.isInstanceOf[MathContext] &&
+      (x.asInstanceOf[MathContext].precision == precision) &&
+      (x.asInstanceOf[MathContext].roundingMode == roundingMode)
```

Comment:
Remove the parentheses, and align everything on the left (so remove two spaces off the last two lines).